ranger-plugin/service/src/main/java/org/apache/hawq/ranger/authorization/RangerHawqPluginResource.java

```diff
@@ -45,8 +47,43 @@
     /**
      * Constructor. Creates a new instance of the resource that uses <code>RangerHawqAuthorizer</code>.
      */
-    public RangerHawqPluginResource() {
+    public RangerHawqPluginResource()
+    {
+        // set UserGroupInformation under kerberos authentication
+        if (Utils.getAuth().equals("kerberos"))
+        {
+            Configuration conf = new Configuration();
+            conf.set("hadoop.security.authentication", "kerberos");
+            UserGroupInformation.setConfiguration(conf);
+
+            String prin = Utils.getPrincipal();
+            String keytab = Utils.getKeytab();
+
+            if ( !prin.equals("") && !keytab.equals("") )
+            {
+                try
+                {
+                    UserGroupInformation.loginUserFromKeytab(prin, keytab);
+                }
+                catch (Exception e)
+                {
+                    LOG.warn(String.format("loginUserFromKeytab failed, user[%s], keytab[%s]", prin, keytab));
+                }
+            }
+        }
+
+        try
+        {
+            UserGroupInformation user = UserGroupInformation.getLoginUser();
+            LOG.info(String.format("login user: %s", user));
```

Comment:
this should be debug level, we do not want to log into log file every time request is processed normally